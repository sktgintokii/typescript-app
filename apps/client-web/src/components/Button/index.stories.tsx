import { action } from '@storybook/addon-actions'
import { text, withKnobs } from '@storybook/addon-knobs'
import * as React from 'react'
import Button from '.'

export default {
  component: Button,
  decorators: [withKnobs],
  title: 'Base/Button'
}

export const standard = () => (
  <Button onClick={action('onClick')}>{text('Label', 'Click Me!')}</Button>
)
