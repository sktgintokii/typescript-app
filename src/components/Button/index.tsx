import React, { ReactNode } from 'react'

export interface ButtonProps {
  onClick?: () => void
  children?: ReactNode
}

const Button: React.SFC<ButtonProps> = ({ onClick, children }) => {
  return <button onClick={onClick}>{children}</button>
}

export default Button
