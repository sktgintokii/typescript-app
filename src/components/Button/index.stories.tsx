import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text } from '@storybook/addon-knobs'
import Button from '.';

const stories = storiesOf('Components', module);
stories.addDecorator(withKnobs);

stories.add(
  'Button',
  () =>
    <Button onClick={action('onClick')}>
      {text('Label', 'Click Me!')}
    </Button>,
);
