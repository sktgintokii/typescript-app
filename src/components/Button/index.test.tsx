import React from 'react'
import renderer from 'react-test-renderer';
import Button from '.'

describe('Button', () => {
  test('Match snapshot', () => {
    const component = renderer.create(
      <Button>Click Me</Button>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
