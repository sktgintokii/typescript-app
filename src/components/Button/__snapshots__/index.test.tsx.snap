// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Match snapshot 1`] = `
<button>
  Click Me
</button>
`;
